src/app/onboarding/views/WelcomeView.spec.ts

```diff
@@ -17,7 +17,6 @@ describe('WelcomeView.vue', () => {
       return merge({
         body: {
           environment: 'universal',
-          mode: 'standalone',
```

Comment:
**Issue**: Until we actually use state like mode from the index.html file, I’m not okay with removing the test coverage for the API-based source of data. We can always do that after we switched (or while we do so) to that approach.

Reply:
We barely have any tests over these values at the moment, and the test here only tests a design element for the Welcome screen in onboarding, which in the scheme of things is a bit 🤷 .

Considering there was no test to check what the snapshot was when this was in non-standalone mode, I'm not totally sure that this is worth blocking on, especially considering the importance of getting this work done.

I noticed you've started expanded on these here https://github.com/kumahq/kuma-gui/pull/1373 but using the old API based approach that I've been working on removing for quite a while now, and considering:

> Once we’re actually using this mechanism, it should be trivial to update the testing approach.

1. We are using this mechanism.
2. You said it in the above PR it should be trivial to update.

Maybe it's best to update it over there in https://github.com/kumahq/kuma-gui/pull/1373?

I agree that testing these things in other parts of the code base is important, so let me know what you think.

Reply:
P.S. I also noticed today whilst reviewing PRs that where merged whilst I was away that quite a lot of test coverage for things like loading and error states where removed in a few of the PRs that went in other the past few weeks. I would be more "not ok" with us removing those, are we definitely sure all our loading/error states are fine?

Reply:
My point was really more general: Since we’re not using the non-API-based approach yet, the tests are more valuable as they’re currently written because they match the reality of our application in use.

I just don’t think we should switch tests to a new approach that isn’t actually in use, yet. That should happen when the change is made, not before. That’s also why I don’t want to update https://github.com/kumahq/kuma-gui/pull/1373 to use a different approach because that wouldn’t test what we’re actually using.

What’s the problem with updating the tests when we actually start using this different approach?

Reply:
> My point was really more general

If its more general then should it be blocking this PR, especially considering the triviality of the test.

> What’s the problem with updating the tests when we actually start using this different approach?

Well you already have that underway right? You said it was trivial to update to use the new approach, seeing as you've known for a while now that we are using this new approach following all the discussions we've had and the PRs that have been merged, I think it would be better to continue and build on the work you have over in #1373 especially considering you think it will be trivial to add.

Reply:
I think expecting our tests to run against the mechanism we actually use is a pretty sensible one to make. This test isn’t relevant enough to block this PR over it so I will resolve this conversation, but I will not change my approach in https://github.com/kumahq/kuma-gui/pull/1373 (considering also that the tests break once I try to switch the approach).

> Do you have any comment on:

Not really, it’s not related to this conversation and our coverage for the tests that I removed is good.